### ISSUE TITLE
Plotting options doesn't work properly for plotly backend

### ISSUE THREAD

**Issue by HuangJunye** [pyviz/holoviews#3698]
I found many options in plots doesn't work in plotly backend. Matplotlib backend generally show desirable output, while bokeh sometimes fails too.

Tested:

|              | plotly                  | bokeh    | matplotlib |
|--------------|-------------------------|----------|------------|
| ticks = n    | N                       | Y        | Y          |
| ticks = list | Y                       | Y        | Y          |
| logx         | xlim become (0,10^xmax) | no ticks | Y          |
| logy         | ylim become (0,10^ymax) | Y        | Y          |
| xaxis        | N                       | Y        | Y          |
| yaxis        | N                       | Y        | Y          |
| xlabel       | Y                       | Y        | Y          |
| ylabel       | Y                       | Y        | Y          |
| xrotation    | N                       | Y        | Y          |
| yrotation    | N                       | Y        | Y          |

```python
# imports
import numpy as np
import pandas as pd
import holoviews as hv
from holoviews import opts
hv.extension('plotly', 'bokeh', 'matplotlib')

# plots some points of exponential function
xs = np.linspace(0, np.pi*2)
points = hv.Points(np.exp(xs))

# a list of ploting options
axes_opts = [
             ('N xticks', {'xticks': 3}),
             ('N yticks', {'yticks': 3}),
             ('List xticks', {'xticks': [0, 5, 10, 20, 50, 100]}),
             ('List yticks', {'yticks': [0, 50, 100, 200, 500, 1000]})
]
# plot and export plots using different backends
backend = 'plotly' # or 'bokeh', 'matplotlib'
hv.output(backend=backend)
layout = hv.Layout([points.relabel(group=group).options(**opts)
           for group, opts in axes_opts]).cols(2)
hv.save(layout, 'layout-%s-log.png' % backend, dpi=300)
```
![layout-plotly-ticks](https://user-images.githubusercontent.com/7631333/57348393-78577800-7188-11e9-865d-cab720716065.png)
![layout-bokeh-ticks](https://user-images.githubusercontent.com/7631333/57348402-7ee5ef80-7188-11e9-829e-c7746c9dac79.png)
![layout-matplotlib-ticks](https://user-images.githubusercontent.com/7631333/57348411-84433a00-7188-11e9-9087-2f2151a7e03f.png)

```python
axes_opts = [
             ('logx', {'logx': True}),
             ('logy', {'logy': True}),
]
```
![layout-plotly-log](https://user-images.githubusercontent.com/7631333/57348487-cbc9c600-7188-11e9-9663-6760e5133ceb.png)
![layout-bokeh-log](https://user-images.githubusercontent.com/7631333/57348495-d08e7a00-7188-11e9-9a9b-f1452c44a0ef.png)
![layout-matplotlib-log](https://user-images.githubusercontent.com/7631333/57348499-d4ba9780-7188-11e9-8fb1-04dca5b4eaa8.png)

```python
axes_opts = [
             ('xaxis', {'xaxis':'top'}),
             ('yaxis', {'yaxis':'right'}),
             ('xlabel', {'xlabel':'xlabel'}),
             ('ylabel', {'ylabel':'ylabel'}),
             ('xrotation', {'xrotation':90}),
             ('yrotation', {'yrotation':90}),
]
```
![layout-plotly-axes](https://user-images.githubusercontent.com/7631333/57348519-ee5bdf00-7188-11e9-82de-5ab9e04a4f0b.png)
![layout-bokeh-axes](https://user-images.githubusercontent.com/7631333/57348521-f287fc80-7188-11e9-92e5-a1d2d2bd4013.png)
![layout-matplotlib-axes](https://user-images.githubusercontent.com/7631333/57348531-f87ddd80-7188-11e9-9c08-8777ce7e6dd7.png)

**Comment by HuangJunye** [comment]
I also found that dpi in hv.save only works for matplotlib backend.

**Comment by jbednar** [comment]
Note that I would only expect `dpi` to work for the matplotlib backend, as Bokeh options are specified in pixels directly, not inches.

**Comment by philippjfr** [comment]
>Note that I would only expect dpi to work for the matplotlib backend, as Bokeh options are specified in pixels directly, not inches.

True, although it would be nice if plotly and bokeh png exports had options to control the dpi (they currently don't). P.S.: GitHub seems to be very confused about timezones today.

**Comment by HuangJunye** [comment]
@philippjfr I find Plotly generally works better than bokeh most of the time. For example, Plotly can automatically recognise date columns as time series while bokeh doesn't and it will show every row as a tick label which is a mess. The hover tool in Plotly is included by default and exported images do not contain the annoying tool bar. I think bokeh should do the same.

**Comment by HuangJunye** [comment]
@jbednar @philippjfr Is there no way to control dpi in Plotly or bokeh backend in holoviews currently? Is there any workaround by importing Plotly or bokeh together with holoviews? Perhaps I should put this in a separate issue.

**Comment by LunarLanding** [comment]
Just tried logy with plotly, still an issue. Any tips regarding which code is related to this?

**Comment by pepijndevos** [comment]
I just ran into this issue where indeed the plot bounds are wrong on Plotly when using log axis. But, when you press autoscale inside the plot, the correct bounds are shown

![newplot(12)](https://user-images.githubusercontent.com/168609/162918405-6880391b-7768-425a-ac0f-ebf20c160ba3.png)
![newplot(13)](https://user-images.githubusercontent.com/168609/162918348-ffd3562c-1dbb-4561-ab51-532c1e1b69dc.png)

**Comment by jbednar** [comment]
For a while we had one of the main Plotly developers maintaining the Plotly backend, but he's moved on, and the core HoloViews development team currently only uses Plotly for a very few plot types (typically 3D) not supported by Bokeh. So this is definitely a case where Plotly users in the community can step in by looking at the Plotly definitions in holoviews/holoviews/plotting/plotly/ and fix missing or broken options. There's no reason the Plotly backend can't be as well supported as the Bokeh one!
